src/main/java/org/mitre/caasd/commons/YyyyMmDd.java

```diff
@@ -136,4 +141,31 @@ public static YyyyMmDd newestOf(Collection<YyyyMmDd> dates) {
         checkNotNull(dates);
         return dates.stream().max(Comparator.naturalOrder()).orElse(null);
     }
+
+
+    private static final String DATE_FORMAT_ERROR = "Date not specified as YYYY-MM-DD format";
+
+    /** Throw an IllegalArgumentException if the input String is not in the YYYY-MM-DD format. */
+    public static void verifyYearMonthDayFormat(String date) {
```

Comment:
Can we not do this with a `DateTimeFormatter`?

Reply:
You'd think so right.  I am not aware of how to cleanly do this with a ISO_LOCAL_DATE formatter.  I could be missing something obvious.  But manually implementing this wasn't my first choice.

Reply:
You can..

```
        try {
            ISO_LOCAL_DATE.parse(date);
        } catch(DateTimeParseException dtpe) {
            throw new IllegalArgumentException("ISO_LOCAL_DATE cannot parse: " + date);
        }
```

Reply:
I rewrote the commit to add the above code